File: models/sponsor.py
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

from odoo import models, fields, api
from odoo.exceptions import ValidationError 

class sponsor(models.Model):
    _name = 'ofc_odoo.sponsor'
    name = fields.Char(required = True)
    email = fields.Char()
    state = fields.Boolean()
    date = fields.Datetime()
    phone = fields.Integer()
    adType = fields.Char()
    event = fields.Many2many('ofc_odoo.event', string ="Events")
    admin = fields.Many2one('res.users', required = True, ofc_admin=True)
    
        
    @api.constrains('name')
    def _validate_lenght_name(self):
        if len(str(self.name)) > 30:
            raise ValidationError("The max name length is 30")
    
    @api.onchange('phone')
    def _check_phone(self):
        if self.phone < 0:
            return{
            'warning':{
                'title': "Incorrect phone length",
                'message': "The phone may not be negative",
            }
    }
    
    @api.constrains('date')    
    def _validate_date(self):
        date_today = fields.Date().today()
        if self.date > date_today:
            raise ValidationError("The date is not equals today")

File: java/entities/Sponsor.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package entities;

import com.fasterxml.jackson.annotation.JsonFormat;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import java.io.Serializable;
import java.util.Date;
import java.util.List;
import javax.persistence.Entity;
import javax.persistence.EnumType;
import javax.persistence.Enumerated;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.ManyToMany;
import javax.persistence.ManyToOne;
import javax.persistence.Table;
import javax.persistence.Temporal;
import javax.persistence.TemporalType;
import javax.xml.bind.annotation.XmlRootElement;

/**
 *
 * @author Elias
 */
@Entity
@Table(name = "SPONSORS", schema = "OFC_DB")
@XmlRootElement
public class Sponsor implements Serializable {

  private static final long serialVersionUID = 1L;
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Integer id;

    private String name;

    private Integer phone;

    private String email;

    @Temporal(TemporalType.TIMESTAMP)
    @JsonSerialize(as=Date.class)
    @JsonFormat(shape=JsonFormat.Shape.STRING, pattern="yyyy-MM-dd'T'HH:mm")
    private Date date;

    private Boolean status;
    
    @ManyToMany
    private List<Event> events;
    
    @ManyToOne
    private Admin admin;

    @Enumerated(EnumType.ORDINAL)
    private AdType ad;

     public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }
 
    public void setEventos(List<Event> eventos) {
        this.events = eventos;
    }

    public List<Event> getEventos() {
        return events;
    }


    public Sponsor() {
        super();
    }


    public void setNombre(String nombre) {
        this.name = nombre;
    }

    public String getNombre() {
        return name;
    }

    public void setTelefono(Integer telefono) {
        this.phone = telefono;
    }

    public Integer getTelefono() {
        return phone;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public String getEmail() {
        return email;
    }

    public void setAnuncio(AdType anuncio) {
        this.ad = anuncio;
    }

    public AdType getAnuncio() {
        return ad;
    }

    public void setFecha(Date fecha) {
        this.date = fecha;
    }

    public Date getFecha() {
        return date;
    }

    public void setEstado(Boolean estado) {
        this.status = estado;
    }

    public Boolean getEstado() {
        return status;
    }

    @Override
    public int hashCode() {
        int hash = 0;
        hash += (id != null ? id.hashCode() : 0);
        return hash;
    }

    @Override
    public boolean equals(Object object) {
        // TODO: Warning - this method won't work in the case the id fields are not set
        if (!(object instanceof Sponsor)) {
            return false;
        }
        Sponsor other = (Sponsor) object;
        if ((this.id == null && other.id != null) || (this.id != null && !this.id.equals(other.id))) {
            return false;
        }
        return true;
    }

    @Override
    public String toString() {
        return "entities.User1[ id=" + id + " ]";
    }
    
    
}
